core/amber/src/main/python/core/architecture/packaging/output_manager.py

```diff
@@ -73,6 +94,17 @@ def get_port(self, port_id=None) -> WorkerPort:
     def get_output_channel_ids(self):
         return self._channels.keys()
 
+    def save_tuple_to_storage_if_needed(self, amber_tuple: Tuple, port_id=None) -> None:
+        if port_id is None:
+            for writer_thread in self._port_result_writers.values():
+                writer_thread.put_tuple(amber_tuple)
+        elif port_id in self._port_result_writers.keys():
+            self._port_result_writers[port_id].put_tuple(amber_tuple)
```

Comment:
can this logic be put into the writer thread? Output manager feels like a weird place for this logic.

Reply:
Yes, done